service/src/main/java/bio/terra/workspace/app/controller/Alpha1ApiController.java

```diff
@@ -0,0 +1,238 @@
+              .resourceType(optResource.map(r -> r.getResourceType().toApiModel()).orElse(null))
+              .resource(
+                  optResource.map(r -> apiResourceFromWsmResource(r, gcpProjectId)).orElse(null));
+      apiJobList.add(apiJob);
+    }
+
+    ApiEnumerateJobsResult result =
+        new ApiEnumerateJobsResult()
+            .pageToken(enumeratedJobs.getPageToken())
+            .totalResults(enumeratedJobs.getTotalResults())
+            .results(apiJobList);
+
+    return new ResponseEntity<>(result, HttpStatus.OK);
+  }
+
+  // Convert a WsmResource into the API format for enumeration
+  @VisibleForTesting
+  public ApiResourceUnion apiResourceFromWsmResource(WsmResource wsmResource, String gcpProjectId) {
+
+    var union = new ApiResourceUnion();
+    switch (wsmResource.getStewardshipType()) {
+      case REFERENCED:
+        ReferencedResource referencedResource = wsmResource.castToReferencedResource();
+        switch (wsmResource.getResourceType()) {
+          case BIG_QUERY_DATASET:
+            {
+              ReferencedBigQueryDatasetResource resource =
+                  referencedResource.castToBigQueryDatasetResource();
+              union.gcpBqDataset(resource.toApiResource());
+              break;
+            }
+          case BIQ_QUERY_DATA_TABLE:
+            {
+              ReferencedBigQueryDataTableResource resource =
+                  referencedResource.castToBigQueryDataTableResource();
+              union.gcpBqDataTable(resource.toApiResource());
+              break;
+            }
+          case DATA_REPO_SNAPSHOT:
+            {
+              ReferencedDataRepoSnapshotResource resource =
+                  referencedResource.castToDataRepoSnapshotResource();
+              union.gcpDataRepoSnapshot(resource.toApiResource());
+              break;
+            }
+
+          case GCS_BUCKET:
+            {
+              ReferencedGcsBucketResource resource = referencedResource.castToGcsBucketResource();
+              union.gcpGcsBucket(resource.toApiModel());
+              break;
+            }
+
+          case GCS_OBJECT:
+            {
+              ReferencedGcsObjectResource resource = referencedResource.castToGcsObjectResource();
+              union.gcpGcsObject(resource.toApiModel());
+              break;
+            }
+
+          default:
+            throw new InternalLogicException(
+                "Unknown referenced resource type: " + wsmResource.getResourceType());
+        }
+        break; // referenced
+
+      case CONTROLLED:
+        ControlledResource controlledResource = wsmResource.castToControlledResource();
+        switch (wsmResource.getResourceType()) {
+          case AI_NOTEBOOK_INSTANCE:
+            {
+              ControlledAiNotebookInstanceResource resource =
+                  controlledResource.castToAiNotebookInstanceResource();
+              union.gcpAiNotebookInstance(resource.toApiResource(gcpProjectId));
```

Comment:
Not sure. If projectId is null it would mean that someone passed an invalid resource as a parameter into a flight. I am reticent to throw, since it would make any enumeration touching that flight fail completely. OTOH, if there are errors like that with corrupt resources, it would be good to log in a way that would alert since it means we have (or had) a bug. What do you think?

@zloery do you have a perspective on this one?

Reply:
I'd rather not break every time we enumerate a bugged flight - I think a better way to handle this would be to log at error level and notify people with something like GCP's log-based alerting (PF-1247), which we've also looked at adding in a few other places.

Reply:
I coded a check with logging on null in a subroutine, so we can easily change our minds about this.